src/AElf.Runtime.WebAssembly/Executive.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System.Text.Json;
+using AElf.Kernel;
+using AElf.Kernel.SmartContract;
+using AElf.Kernel.SmartContract.Infrastructure;
+using AElf.Types;
+using Google.Protobuf;
+using Google.Protobuf.Reflection;
+using Google.Protobuf.WellKnownTypes;
+using NBitcoin.DataEncoders;
+using Solang;
+using Solang.Extensions;
+using Wasmtime;
+
+namespace AElf.Runtime.WebAssembly;
+
+public class Executive : IExecutive
+{
+    public IReadOnlyList<ServiceDescriptor> Descriptors { get; }
+    public Hash ContractHash { get; set; }
+    public Timestamp LastUsedTime { get; set; }
+    public string ContractVersion { get; set; }
+
+    private readonly WebAssemblyRuntime _webAssemblyRuntime;
+    private readonly SolangABI _solangAbi;
+
+    private IHostSmartContractBridgeContext _hostSmartContractBridgeContext;
+    private ITransactionContext CurrentTransactionContext => _hostSmartContractBridgeContext.TransactionContext;
+
+    public Executive(IExternalEnvironment ext, CompiledContract compiledContract)
+    {
+        var wasmCode = compiledContract.WasmCode.ToByteArray();
+        _solangAbi = JsonSerializer.Deserialize<SolangABI>(compiledContract.Abi)!;
+        _webAssemblyRuntime = new WebAssemblyRuntime(ext, wasmCode);
+        ContractHash = HashHelper.ComputeFrom(wasmCode);
+        ContractVersion = "Unknown solidity version.";
+    }
+
+    public IExecutive SetHostSmartContractBridgeContext(IHostSmartContractBridgeContext smartContractBridgeContext)
+    {
+        _hostSmartContractBridgeContext = smartContractBridgeContext;
+        _webAssemblyRuntime.SetHostSmartContractBridgeContext(smartContractBridgeContext);
+        return this;
+    }
+
+    public Task ApplyAsync(ITransactionContext transactionContext)
+    {
+        _hostSmartContractBridgeContext.TransactionContext = transactionContext;
+        Execute();
+        return Task.CompletedTask;
+    }
+
+    private void Execute()
+    {
+        var s = CurrentTransactionContext.Trace.StartTime = TimestampHelper.GetUtcNow().ToDateTime();
+
+        var transactionContext = _hostSmartContractBridgeContext.TransactionContext;
+        var transaction = transactionContext.Transaction;
+
+        var isCallConstructor = transaction.MethodName == "deploy";
+
+        if (isCallConstructor && (_hostSmartContractBridgeContext.Origin != null ||
+                                  _hostSmartContractBridgeContext.Origin == _hostSmartContractBridgeContext.Sender))
+        {
+            transactionContext.Trace.ExecutionStatus = ExecutionStatus.Prefailed;
+            transactionContext.Trace.Error = "Cannot execute constructor.";
+            return;
+        }
+
+        var selector = _solangAbi.GetSelector(transaction.MethodName);
```

Comment:
We don't need to do this conversion. The compiled `wasm` code will do this automatically. I think on the transaction level we just need to differentiate whether the transaction is a contract call or contract creation.

Reply:
If method is not required, maybe we can leave it empty.